### ISSUE TITLE
Add tf.metrics.std or tf.metrics.var

### ISSUE THREAD

**Issue by EmanueleGhelfi** [tensorflow/tensorflow#24192]
**System information**
- TensorFlow version (you are using): 1.12
- Are you willing to contribute it (Yes/No): Maybe, it depends on the complexity

**Describe the feature and the current behavior/state.**
Currently the module tf.metrics does not contain tf.metrics.std or tf.metrics.var for the calculation of standard deviation and variance over multiple batches, respectively. It would be very useful to have these type of metrics.

**Will this change the current api? How?**
This will change the current api by adding two more apis: tf.metrics.std  and  tf.metrics.var

**Who will benefit with this feature?**
Everyone who wants to calculate statistics over multiple batches.

**Any Other info.**
Currently it is possible to use tf.nn.moments and perform some basic math in order to calculate the std over multiple batches. However tf.metrics is a simple way to obtain the same result.

**Comment by ymodak** [comment]
You can use [tf.nn.moments](https://www.tensorflow.org/api_docs/python/tf/nn/moments) to calculate variance and mean.

**Comment by EmanueleGhelfi** [comment]
As I said, it is possible to use tf.nn.moments but it calculates the mean and the variance only for a batch and it does not consider multiple batches. Tf.metrics provides a standard way to calculate metrics over multiple batches but tf.metrics.var is not present.

**Comment by eralvc** [comment]
> **System information**
> 
> * TensorFlow version (you are using): 1.12
> * Are you willing to contribute it (Yes/No): Maybe, it depends on the complexity
> 
> **Describe the feature and the current behavior/state.**
> Currently the module tf.metrics does not contain tf.metrics.std or tf.metrics.var for the calculation of standard deviation and variance over multiple batches, respectively. It would be very useful to have these type of metrics.
> 
> **Will this change the current api? How?**
> This will change the current api by adding two more apis: tf.metrics.std and tf.metrics.var
> 
> **Who will benefit with this feature?**
> Everyone who wants to calculate statistics over multiple batches.
> 
> **Any Other info.**
> Currently it is possible to use tf.nn.moments and perform some basic math in order to calculate the std over multiple batches. However tf.metrics is a simple way to obtain the same result.

I have a idea: It is possible to use tf.metric.mean(tensor^2) calculate the average of square,  and use tf.metric.mean(tensor) to calculate the average. 
Then you can use the other form of variance:
var(X) =1/N *[ (x1- avg)^2+(x2-avg)^2+...+(xn-avg)^2]
           =1/N *[ (x1^2 - 2*x1*avg+ avg^2) + (x2^2 - 2*x2*avg+ avg^2) +...+(xn^2 - 2*xn*avg+ avg^2) ]
           = 1/N *[(x1^2 + x2^2 + ... + xn^2) - 2 *(x1+x2+...+xn)*avg + N * avg^2]
           = 1/N *[(x1^2 + x2^2 + ... + xn^2)  - 2 * N * avg *avg + N*avg^2]
           =1/N *[(x1^2 + x2^2 + ... + xn^2)  -  N*avg^2]
           =1/N *(x1^2 + x2^2 + ... + xn^2) - avg^2

**Comment by rmothukuru** [comment]
@EmanueleGhelfi,
The solution mentioned in [this comment](https://github.com/tensorflow/tensorflow/issues/24192#issuecomment-565336012) can be implemented either in the [**`on_epoch_begin`**](https://www.tensorflow.org/api_docs/python/tf/keras/callbacks/Callback#on_epoch_begin) or in the [**`on_epoch_end`**](https://www.tensorflow.org/api_docs/python/tf/keras/callbacks/Callback#on_epoch_end) function of [Custom Callback](https://www.tensorflow.org/guide/keras/custom_callback#usage_of_logs_dict).

Or you can use [Custom Metrics](https://www.tensorflow.org/guide/keras/train_and_evaluate#custom_metrics) to design the metrics of your choice.

Thanks!
